client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { Route, Redirect } from 'react-router-dom';
+type Props = {
+  children: JSX.Element;
+  path: string;
+  exact: boolean;
+};
+const ProtectedRoute: React.FC<Props> = ({ children, ...restOfProps }) => {
+  const { loggedInUser } = useAuth();
+  let isLoggedInAndValid = false;
+
+  if (loggedInUser && loggedInUser.username && loggedInUser.email) {
+    isLoggedInAndValid = true;
+  }
```

Comment:
Do you think we can simplify this to:
```suggestion
  const isLoggedInAndValid =
    loggedInUser && loggedInUser.username && loggedInUser.email;
```

Reply:
Makes sense, I should have figured out. Thanks so much!